python_backend/backend.py

```diff
@@ -0,0 +1,35 @@
+import json
+from haversine import inverse_haversine, Direction, Unit
+
+# Any interesting state should be stored in a singleton of this object
```

Comment:
The idea is that this code should be structured in a way that could make it useful as a standalone Python library, or called from notebooks. An example real structure could take in OSM input data, parse into a network, and store things as state. The web frontend makes a `Backend` singleton and calls methods on it.